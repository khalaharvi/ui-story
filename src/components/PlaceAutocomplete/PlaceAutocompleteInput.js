import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';
import _get from 'lodash.get';

import { Plus as PlusIcon } from 'icons';
import { cleanButton } from '../../styles';

const InputContainer = styled('div')({
  position: 'relative'
});

const InputButton = styled('button')(({ theme, hidden }) => ({
  ...cleanButton,
  display: hidden ? 'none' : 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  position: 'absolute',
  right: 5,
  top: '50%',
  transform: 'translateY(-50%)',
  height: 40,
  width: 40,

  ':hover svg': {
    fill: _get(theme, 'text.primary')
  },

  svg: {
    display: 'block',
    width: 14,
    height: 14,
    fill: _get(theme, 'text.secondary')
  }
}));

const PlaceAutocompleteInput = ({
  canReset,
  onReset,
  ...props
}) => (
  <InputContainer>
    <input {...props} />
    <InputButton type="button" hidden={!canReset} onClick={onReset}>
      <PlusIcon rotate />
    </InputButton>
  </InputContainer>
);

PlaceAutocompleteInput.propTypes = {
  onReset: PropTypes.func,
  canReset: PropTypes.bool
};

PlaceAutocompleteInput.defaultProps = {
  canReset: false
};

export default PlaceAutocompleteInput;
