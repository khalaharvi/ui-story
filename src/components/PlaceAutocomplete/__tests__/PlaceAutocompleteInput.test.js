import React from 'react';
import { shallow } from 'enzyme';

import PlaceAutocompleteInput from '../PlaceAutocompleteInput';

describe('<PlaceAutocompleteInput />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<PlaceAutocompleteInput />);
    expect(wrapper).toMatchSnapshot();
  });
});
