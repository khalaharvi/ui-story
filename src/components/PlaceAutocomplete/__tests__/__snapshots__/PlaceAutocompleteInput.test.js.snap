// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaceAutocompleteInput /> renders correctly 1`] = `
<Styled(div)>
  <input />
  <Styled(button)
    hidden={true}
    type="button"
  >
    <Plus
      rotate={true}
    />
  </Styled(button)>
</Styled(div)>
`;
